src/vs/workbench/contrib/chat/browser/chatViewPane.ts

```diff
@@ -81,6 +85,20 @@ export class ChatViewPane extends ViewPane implements IViewWelcomeDelegate {
 		// View state for the ViewPane is currently global per-provider basically, but some other strictly per-model state will require a separate memento.
 		this.memento = new Memento('interactive-session-view-' + CHAT_PROVIDER_ID, this.storageService);
 		this.viewState = this.memento.getMemento(StorageScope.WORKSPACE, StorageTarget.MACHINE);
+		this.titleController = this._register(new ChatViewTitleController(
+			options.id,
+			{
+				getCurrentModel: () => this._widget?.viewModel?.model,
+				updatePrimaryTitle: title => this.updateTitle(title),
+				requestLayout: () => {
+					if (this._lastLayout) {
+						this.layoutBody(this._lastLayout.height, this._lastLayout.width);
```

Comment:
This delegate pattern is fine, but another way to think about this one would be to expose a `onDidChangeHeight` event on the title part, and listen to that and react to it in the view pane.